mesonbuild/dependencies/pkgconfig.py

```diff
@@ -277,6 +277,30 @@ def _search_libs(self, out: str, out_raw: str) -> T.Tuple[T.List[str], T.List[st
             elif lib.startswith('-L'):
                 # We already handled library paths above
                 continue
+            elif lib.startswith('-l:'):
+                # see: https://stackoverflow.com/questions/48532868/gcc-library-option-with-a-colon-llibevent-a
+                # also : See the documentation of -lnamespec | --library=namespec in the linker manual  
+                #                     https://sourceware.org/binutils/docs-2.18/ld/Options.html
+                
+                # Don't resolve the same -l:libfoo.a argument again
+                if lib in libs_found:
+                    continue
+                libfilename = lib[3:]
+                foundname = None
+                for libdir in libpaths:
+                    target = os.path.join(libdir, libfilename)
+                    if os.path.exists(target):
+                        foundname = target
+                        break
+                if foundname is None:
+                    if lib in libs_notfound:
+                        continue
+                    if self.static:
+                        mlog.warning('Static library {!r} not found for dependency {!r}, may '
+                                    'not be statically linked'.format(libfilename, self.name))
```

Comment:
Do libraries specified with `-l:filename` *have* to be static, or does a `-l:foobar.so.0.0.1` also work here? If this is the case, this has to be adjusted.

Reply:
Good question! It does not, it's just treated as a filename literal rather than performing heuristic lookup while preferring the heuristics for shared or static libraries, depending.